content/docs/03-ai-sdk-core/10-generating-structured-data.mdx

```diff
@@ -111,3 +111,44 @@ const result = await generateObject({
   prompt: 'Generate a fake user profile for testing.',
 });
 ```
+
+## Error Handling
+
+When you use `generateObject`, errors are thrown when the model fails to generate property JSON (`JSONParseError`)
+or when the generated JSON does not match the schema (`TypeValidationError`).
+Both error types contain additional information, e.g. the generated text or the invalid value.
+
+You can use this to e.g. design a function that safely process the result object and also returns values in error cases:
+
+```ts
+import { openai } from '@ai-sdk/openai';
+import { JSONParseError, TypeValidationError, generateObject } from 'ai';
+import { z } from 'zod';
+
+async function example<T>(
```

Comment:
should we name this something more in line with what it's doing?

Reply:
good idea